Products/ZenModel/migrate/editServicesStartUp.py

```diff
@@ -0,0 +1,129 @@
+##############################################################################
+#
+# Copyright (C) Zenoss, Inc. 2019, all rights reserved.
+#
+# This content is made available according to terms specified in
+# License.zenoss under the directory where your Zenoss product is installed.
+#
+##############################################################################
+
+from collections import namedtuple
+
+import logging
+import string
+import Migrate
+import servicemigration as sm
+from Products.ZenModel.ZMigrateVersion import SCHEMA_MAJOR, SCHEMA_MINOR, SCHEMA_REVISION
+
+log = logging.getLogger("zen.migrate")
+sm.require("1.1.14")
+
+
+class EditServicesStartUp(Migrate.Step):
+    "Edit service command"
+
+    version = Migrate.Version(SCHEMA_MAJOR, SCHEMA_MINOR, SCHEMA_REVISION) 
+
+    def cutover(self, dmd):
+        try:
+            ctx = sm.ServiceContext()
+        except sm.ServiceMigrationError:
+            log.info("Couldn't generate service context, skipping.")
+            return
+
+        def update_service(collection):
+            services = filter(lambda s: s.name in collection, ctx.services)
+            for service in services:
+                service.startup = command.format(service.name)
+                service.runAs = "zenoss"
+
+        service_names = ("zencommand", "zenmodeler", "zenperfsnmp", "zenping", "zenprocess", "zenstatus")
+        command = "/opt/zenoss/bin/{} run -c --logfileonly --workers {{.Instances}} --workerid $CONTROLPLANE_INSTANCE_ID --monitor {{(parent .).Name}}"
```

Comment:
You need to escape the double `{` braces in `command`.  Python's `format` method will see `{` as its own formatting directive and remove a `{}` brace pair, which then leaves you with an incorrectly formatted command, e.g.
```
/opt/zenoss/bin/zencommand run -c --logfileonly --workers {.Instances} --workerid $CONTROLPLANE_INSTANCE_ID --monitor {(parent .).Name}
```
Note the single `{` braced `.Instances` and `(parent .).Name` parameters.

Of course you could use Python's `%` string interpolation to avoid escaping the double `{` fields.